course_content/notebooks/dask_intro.ipynb

```diff
@@ -0,0 +1,216 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# An Introduction to Dask\n",
+    "\n",
+    "Dask is a flexible parallel computing library for Python. It allows dynamic scheduling of tasks for optimising computations, and includes collections such as arrays and lists for parallel data processing."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Terminology\n",
+    "\n",
+    "### General"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "#### Thread\n",
+    "A thread is the simplest unit of computation.\n",
+    "\n",
+    "#### Process\n",
+    "A process is an isolated computation which consists of one or more threads."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Dask\n",
+    "\n",
+    "#### Dask architecture\n",
+    "Below is the basic structure of how the components of dask interact:\n",
+    "\n",
+    "![](../images/dask_architecture.png)\n"
```

Comment:
This image is good, but doesn't quite tell the whole story! I think we could make things a little clearer by accepting the image as is and adding a little description below, such as:

"""
Dask uses a pool of workers to process tasks specified by the scheduler. The scheduler determines the tasks to be done by intelligently traversing the task graph. The task graph is produced dynamically and automatically passed to the scheduler when using dask functionality in your code. In the case of dask distributed, the task graph is submitted to the scheduler by the client.
"""

Reply:
Thanks, this description has been added.